assignment5/unknownLanguage.py

```diff
@@ -0,0 +1,76 @@
+def unknownLanguage(dictionary):
+    """Given a dictionary of all words in an unknown/invented language, find the
+    alphabet of that language.
+
+    Args:
+        dictionary: a list of words in lexicographic order.
+
+    Returns:
+        the alphabet, an ordered list of characters.
+    """
+    def dfs(v, visited):
+        """
+        Depth-first search with cycle search.
+        """
+        visited[v] = 1
+        for u in adjList[v]:
+            if visited[u] == 0:
+                dfs(u, visited)
+            elif visited[u] == 1:
+                # in that case, we have cycles in our graph =>
+                # it can't be topological sorted => it's not an alphabet
+                raise ValueError("dictionary is not in lexicographic order")
+        visited[v] = 2
+        alphabet.append(v)
+
+    def topologicalSort():
+        """
+        An ordinary topological sort based on dfs.
+        """
+        visited = dict()
+        for v in adjList:
+            visited[v] = 0
+        for v in adjList:
+            if not visited[v]:
+                dfs(v, visited)
+        alphabet.reverse()
+
+
+    if not dictionary:
+        return []
+
+    # define adjacency list where nodes are letters, not nums
+    adjList = dict()
+    for word in dictionary:
+        for let in word:
+            if let not in adjList:
+                adjList[let] = set()
+
+    maxLen = max([len(word) for word in dictionary])
+
+    # bruteforce all possible prefixes by they length
+    for prefixLen in range(maxLen):
+        curPrefix = None
+        prevLetter = ''
+        for word in dictionary:
+            if curPrefix is None:  # so, previous word didn't have proper length
+                # > because we want the NEXT letter after prefix
+                if len(word) > prefixLen:
+                    curPrefix = word[:prefixLen]
+                    prevLetter = word[prefixLen]
+                continue
```

Comment:
Instead of using `continue` here, `elif` would have the same result and reduce the complexity of understanding the code: `continue` and `break` are useful in some cases to make the code simpler to understand, but in some other can make things harder to understand. In this case, using else makes it clearer that curPrefix must not be None in the rest of the code.